mpas_analysis/test/test_climatology.py

```diff
@@ -0,0 +1,228 @@
+        self.assertEqual(regriddedFileName, expectedRegriddedFileName)
+
+    def test_get_observation_climatology_file_names(self):
+        config = self.setup_config()
+        fieldName = 'sst'
+        monthNames = 'JFM'
+        gridFileName = str(self.datadir.join('obsGrid.nc'))
+        componentName = 'ocean'
+        (climatologyFileName, regriddedFileName) = \
+            climatology.get_observation_climatology_file_names(
+                config, fieldName, monthNames, componentName, gridFileName,
+                latVarName='lat', lonVarName='lon')
+        expectedClimatologyFileName = '{}/clim/obs/sst_1.0x1.0degree_' \
+                                      'JFM.nc'.format(self.test_dir)
+        self.assertEqual(climatologyFileName, expectedClimatologyFileName)
+
+        expectedRegriddedFileName = '{}/clim/obs/regrid/sst_1.0x1.0degree_' \
+                                    'to_0.5x0.5degree_' \
+                                    'JFM.nc'.format(self.test_dir)
+        self.assertEqual(regriddedFileName, expectedRegriddedFileName)
+
+    def open_test_ds(self, config, calendar):
+        fileNames = ['{}/timeSeries.0002-{:02d}-01.nc'.format(self.datadir,
+                                                              month)
+                     for month in [1, 2, 3]]
+
+        variableMap = {'mld': ['timeMonthly_avg_tThreshMLD'],
+                       'Time': [['xtime_startMonthly', 'xtime_endMonthly']]}
+        variableList = ['mld']
+
+        ds = open_multifile_dataset(
+            fileNames=fileNames,
+            calendar=calendar,
+            config=config,
+            timeVariableName='Time',
+            variableList=variableList,
+            variableMap=variableMap)
+
+        assert(len(ds.Time) == 3)
+        return ds
+
+    def test_compute_climatology(self):
+        config = self.setup_config()
+        calendar = 'gregorian_noleap'
+        ds = self.open_test_ds(config, calendar)
+
+        assert('month' not in ds.coords.keys())
+        assert('daysInMonth' not in ds.coords.keys())
+
+        # test add_months_and_days_in_month
+        ds = climatology.add_months_and_days_in_month(ds, calendar)
+
+        self.assertArrayEqual(ds.month.values, [1, 2, 3])
+        self.assertArrayEqual(numpy.round(ds.daysInMonth.values), [31, 28, 31])
+
+        # test compute_climatology on a data set
+        monthNames = 'JFM'
+        monthValues = constants.monthDictionary[monthNames]
+        dsClimatology = climatology.compute_climatology(ds, monthValues,
+                                                        calendar)
+
+        assert('Time' not in dsClimatology.dims.keys())
+
+        self.assertEqual(dsClimatology.data_vars.keys(), ['mld'])
+
+        # test compute_climatology on a data array
+        mldClimatology = climatology.compute_climatology(ds.mld, monthValues,
+                                                         calendar)
```

Comment:
> Do you want to make sure mldClimatology and dsClimatology produce similar results? This just tests that Dataset vs DataArray computations work, which this test function implies anyway.

I have updated the test to do this

> Is there some way that we can test the actually computed values, at least in an approximate way? This may be overboard but a climatology is a key computation so this is something to consider.

I think this is also a good idea but I don't know when I will have time to add this capability.  Maybe as a follow-on PR?